crates/external/rpc/src/traits.rs

```diff
@@ -6,20 +6,37 @@ use alloy_primitives::Log;
 use alloy_sol_types::SolEvent;
 use async_trait::async_trait;
 use core::time::Duration;
-use jsonrpsee::core::ClientError;
+use jsonrpsee::{core::ClientError, types::ErrorObjectOwned};
 use kona_interop::{CROSS_L2_INBOX_ADDRESS, ExecutingMessage, SafetyLevel};
 use tokio::time::error::Elapsed;
 
+const UNKNOWN_CHAIN_MSG: &str = "unknown chain: ";
+const MINIMUM_SAFETY_MSG: &str = "does not meet the minimum safety";
 /// Failures occurring during validation of [`ExecutingMessage`]s.
 #[derive(thiserror::Error, Debug)]
 pub enum ExecutingMessageValidatorError {
+    /// Message does not minimum safety level
+    #[error("message does not meet safety level. expected: {expected}, Got: {got}")]
+    MinimumSafety {
+        /// Actual level of the message
+        got: SafetyLevel,
+        /// Level that was passed to supervisor
+        expected: SafetyLevel,
+    },
+    /// Invalid chain
+    #[error("unsupported chain id: {0}")]
+    UnknownChain(u64),
     /// Failure from the [`SupervisorApiClient`] when validating messages.
-    #[error("Supervisor determined messages are invalid: {0}")]
+    #[error("supervisor determined messages are invalid: {0}")]
     SupervisorRpcError(#[from] ClientError),
```

Comment:
let's change this to `RpcClientError(ClientError)`. the name better reflects distinction with server errors, and the `ClientError` should not convert into this variant in case of safety level error.